tests/core/atoms/test_atoms_core.py

```diff
@@ -208,3 +213,18 @@ def test_check_charge_and_spin(os_atoms):
     charge, spin_multiplicity = check_charge_and_spin(atoms, charge=0)
     assert charge == 0
     assert spin_multiplicity == 3
+
+
+def test_perturb():
+    atoms = Atoms("H2", positions=[(0, 0, 0), (0, 0, 0.74)])
+    matrix = [[0.1, 0.1, 0.1], [0.2, 0.2, 0.2]]
+    scale = 0.5
+    perturbed_atoms = perturb(atoms, matrix, scale)
+    assert np.allclose(
+        perturbed_atoms.get_positions(),
+        [[0.05, 0.05, 0.05], [0.1, 0.1, 0.84]],
+        atol=1e-3,
+    ), "Perturbation did not work as expected"
+    assert np.allclose(
+        atoms.get_positions(), [[0, 0, 0], [0, 0, 0.74]], atol=1e-3
+    ), "Original atoms object was modified"
```

Comment:
I'd recommend using the dedicated numpy testing utility for this, namely [`assert_allclose`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html).

It'd be used as

```python
from numpy.testing import assert_allclose

assert_allclose(...) # no need to put an `assert` in front
```